porcupine/plugins/anchors.py

```diff
@@ -0,0 +1,173 @@
+                next_anchor_row = anchor_row
+            else:
+                break
+
+        if not next_anchor_row:
+            return
+        else:
+            self.tab_textwidget.mark_set("insert", f"{str(next_anchor_row)}.0")
+
+        # If cursor is below last row
+        if int(self._get_cursor_index().split(".")[0]) > int(
+            self.tab_textwidget.index(f"@0,{self.tab_textwidget.winfo_height()}").split(".")[0]
+        ):
+            self.tab_textwidget.see("insert")
+
+        # text.see("insert")
+
+        # first_line = int(self._textwidget.index("@0,0").split(".")[0])
+        # last_line = int(
+        #     self._textwidget.index(f"@0,{self._textwidget.winfo_height()}").split(".")[0]
+        # )
+
+        # TODO: If user jumps to the last anchor and then to next, it should jump to the first one in the file.
+
+    def jump_to_previous(self, event: tkinter.Event[tkinter.Misc]) -> None:
+        cursor_row = self._get_cursor_index().split(".")[0]
+        anchor_list = self._get_anchors()
+        anchor_rows = sorted(
+            [
+                int(self.tab_textwidget.index(anchorpoint).split(".")[0])
+                for anchorpoint in anchor_list
+            ]
+        )
+
+        next_anchor_row = 0
+        for anchor_row in anchor_rows:
+            if anchor_row < int(cursor_row):
+                next_anchor_row = anchor_row
+            else:
+                break
+
+        if not next_anchor_row:
+            return
+        else:
+            self.tab_textwidget.mark_set("insert", f"{str(next_anchor_row)}.0")
+
+        # If cursor is above first row
+        if int(self._get_cursor_index().split(".")[0]) < int(
+            self.tab_textwidget.index("@0,0").split(".")[0]
+        ):
+            self.tab_textwidget.see("insert")
+
+        # TODO: Current Bugs:
+        # 1) Jump to previous will delete code on current row if no more anchors above.
+        # 2) Can't commit/push because of git merge weirdness
+
+    # def bind_specific(self, event: tkinter.Event[tkinter.Misc], partial ?) -> None:
+    #     pass
+
+    # def jump_to_specific(self, event: tkinter.Event[tkinter.Misc], partial ?) -> None:
+    #     pass
+
+    def update_linenumbers(self, event: tkinter.Event[tkinter.Misc]) -> None:
+        """
+        Re-draws the anchor points every time the linenumber instance updates
+        (scroll, insertion/deletion of text)
+        """
+        self.prevent_duplicate_anchors()
+
+        anchor_list = self._get_anchors()
+
+        for anchorpoint in anchor_list:
+            row_tag = "line_" + self.tab_textwidget.index(anchorpoint).split(".")[0]
+            try:
+                [row_id] = self.linenumbers.find_withtag(row_tag)
+                row_text = self.linenumbers.itemcget(row_id, "text")
+                self.linenumbers.itemconfigure(row_id, text=row_text + " " + self.anchor_symbol)
+            except ValueError:
+                pass
```

Comment:
Look up `try,except,else` and consider using it here. If `itemcget()` or `itemconfigure()` raises a `ValueError` for whatever reason, you don't want it to get silently ignored.

I would also appreciate a comment like `# line not visible`.